cwms-data-api/src/test/java/cwms/cda/api/project/ProjectLockCatalogHandlerIT.java

```diff
@@ -0,0 +1,141 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.api.project;
+
+import static cwms.cda.api.Controllers.APPLICATION_ID;
+import static cwms.cda.api.Controllers.OFFICE_MASK;
+import static cwms.cda.api.Controllers.PROJECT_ID;
+import static cwms.cda.api.Controllers.TIME_ZONE_ID;
+import static cwms.cda.api.project.ProjectLockHandlerUtil.buildTestProject;
+import static cwms.cda.api.project.ProjectLockHandlerUtil.deleteProject;
+import static cwms.cda.data.dao.DaoTest.getDslContext;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.flogger.FluentLogger;
+import cwms.cda.api.DataApiTestIT;
+import cwms.cda.data.dao.project.ProjectDao;
+import cwms.cda.data.dao.project.ProjectLockDao;
+import cwms.cda.data.dto.project.Project;
+import cwms.cda.formatters.Formats;
+import fixtures.CwmsDataApiSetupCallback;
+import fixtures.TestAccounts;
+import io.restassured.filter.log.LogDetail;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import javax.servlet.http.HttpServletResponse;
+import mil.army.usace.hec.test.database.CwmsDatabaseContainer;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("integration")
+public class ProjectLockCatalogHandlerIT extends DataApiTestIT {
+    private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+
+    public static final String OFFICE = "SPK";
+
+
+    @Test
+    void test_cat_locks() throws SQLException {
+
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+        databaseLink.connection(c -> {
+            DSLContext dsl = getDslContext(c, OFFICE);
+            ProjectDao prjDao = new ProjectDao(dsl);
+            ProjectLockDao lockDao = new ProjectLockDao(dsl);
+
+            String projId = "catLocks";
+            String appId = "catLocks_test";
+            TestAccounts.KeyUser user = TestAccounts.KeyUser.SPK_NORMAL;
+            String userName = user.getName();
+            String officeMask = OFFICE;
+            Project testProject = buildTestProject(OFFICE, projId);
+            prjDao.create(testProject);
+            try {
+                int revokeTimeout = 10;
+
+                lockDao.updateLockRevokerRights(OFFICE, userName, projId, appId + "_1", officeMask, true);
+                lockDao.updateLockRevokerRights(OFFICE, userName, projId, appId + "_2", officeMask, true);
+                String lock1 = lockDao.requestLock(OFFICE, projId, appId + "_1", true, revokeTimeout);
```

Comment:
I just wanted multiple locks so that the catalog type call would return more than one results.  In my test I made them have different applicationId - that was bad and unrealistic.  A more realistic test would have created multiple projects with the same applicationId.  REGI makes use of these Locks and LockRevokerRights.  I don't know of anything else that is using them but @adamkorynta  or @rma-psmorris might have more.

Reply:
As for ordering the acquires and releases to avoid deadlock.  CDA could implement bulk-acquire and bulk-release methods that internally did the ordering.

Reply:
Sounds like a reasonable mitigation for now.

Reply:
But a test should indicate and follow actual usage so I ask that the test be updated to reflect your comment above.

Reply:
Done - changed to make two projects as suggested.

Reply:
sorry I missed your earlier comment. Only REGI uses the Project locks currently.